src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -673,27 +673,29 @@ private BoundStatement BindDeclarationStatementParts(LocalDeclarationStatementSy
             originalBinder.BindForOrUsingOrFixedDeclarations(declarationSyntax, LocalDeclarationKind.UsingVariable, diagnostics, out declarations);
             Debug.Assert(!declarations.IsEmpty);
 
-            BoundTypeExpression declTypeExpr = declarations[0].DeclaredType;
+            BoundLocalDeclaration declaration = declarations[0];
+            TypeSymbol declType = declaration.DeclaredType.Type;
+            BoundLocal declarationLocal = new BoundLocal(declaration.Syntax, declaration.LocalSymbol, null, declType);
```

Comment:
>BoundLocal declarationLocal [](start = 12, length = 27)

Can be moved inside `if (!iDisposableConversion.IsImplicit) { ... }`.

Reply:
Done